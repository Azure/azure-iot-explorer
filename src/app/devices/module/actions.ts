/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import actionCreatorFactory from 'typescript-fsa';
import * as actionPrefixes from '../../constants/actionPrefixes';
import * as actionTypes from '../../constants/actionTypes';
import { ModuleIdentity } from './../../api/models/moduleIdentity';
import { ModuleTwin } from './../../api/models/moduleTwin';

const moduleIdentityCreator = actionCreatorFactory(actionPrefixes.MODULEIDENTITY);
const getModuleIdentitiesAction = moduleIdentityCreator.async<string, ModuleIdentity[]>(actionTypes.GET_MODULE_IDENTITIES);
const addModuleIdentityAction = moduleIdentityCreator.async<ModuleIdentity, ModuleIdentity>(actionTypes.ADD_MODULE_IDENTITY);
const getModuleIdentityTwinAction = moduleIdentityCreator.async<GetModuleIdentityTwinActionParameters, ModuleTwin>(actionTypes.GET_MODULE_IDENTITY_TWIN);
const getModuleIdentityAction = moduleIdentityCreator.async<GetModuleIdentityActionParameters, ModuleIdentity>(actionTypes.GET_MODULE_IDENTITY);
const deleteModuleIdentityAction = moduleIdentityCreator.async<DeleteModuleIdentityActionParameters, void>(actionTypes.DELETE_MODULE_IDENTITY);

export {
    addModuleIdentityAction,
    deleteModuleIdentityAction,
    getModuleIdentitiesAction,
    getModuleIdentityAction,
    getModuleIdentityTwinAction
};

export interface GetModuleIdentityTwinActionParameters {
    moduleId: string;
    deviceId: string;
}

export interface GetModuleIdentityActionParameters {
    moduleId: string;
    deviceId: string;
}

export interface DeleteModuleIdentityActionParameters {
    moduleId: string;
    deviceId: string;
}
