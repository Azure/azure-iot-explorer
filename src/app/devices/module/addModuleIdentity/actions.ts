/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import actionCreatorFactory from 'typescript-fsa';
import { MODULEIDENTITY } from '../../../constants/actionPrefixes';
import { ADD_MODULE_IDENTITY } from '../../../constants/actionTypes';
import { ModuleIdentity } from '../../../api/models/moduleIdentity';

export const addModuleIdentityAction = actionCreatorFactory(MODULEIDENTITY).async<ModuleIdentity, ModuleIdentity>(ADD_MODULE_IDENTITY);
