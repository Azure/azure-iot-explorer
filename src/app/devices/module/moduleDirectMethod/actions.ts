/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import actionCreatorFactory from 'typescript-fsa';
import { DEVICECONTENT } from '../../../constants/actionPrefixes';
import { INVOKE_MODULE_IDENTITY_METHOD } from '../../../constants/actionTypes';

export const invokeModuleDirectMethodAction = actionCreatorFactory(DEVICECONTENT).async<InvokeModuleMethodActionParameters, string>(INVOKE_MODULE_IDENTITY_METHOD);

export interface InvokeModuleMethodActionParameters {
    connectTimeoutInSeconds: number;
    deviceId: string;
    methodName: string;
    moduleId: string;
    payload?: any; // tslint:disable-line:no-any
    responseTimeoutInSeconds: number;
}
