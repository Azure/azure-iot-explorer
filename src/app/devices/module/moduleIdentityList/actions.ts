/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import actionCreatorFactory from 'typescript-fsa';
import { MODULEIDENTITY } from '../../../constants/actionPrefixes';
import { GET_MODULE_IDENTITIES } from '../../../constants/actionTypes';
import { ModuleIdentity } from '../../../api/models/moduleIdentity';

export const getModuleIdentitiesAction = actionCreatorFactory(MODULEIDENTITY).async<string, ModuleIdentity[]>(GET_MODULE_IDENTITIES);
