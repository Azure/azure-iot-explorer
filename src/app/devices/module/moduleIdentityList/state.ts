/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import { Record } from 'immutable';
import { IM } from '../../../shared/types/types';
import { SynchronizationWrapper } from '../../../api/models/synchronizationWrapper';
import { ModuleIdentity } from '../../../api/models/moduleIdentity';
import { SynchronizationStatus } from '../../../api/models/synchronizationStatus';

export interface ModuleIndentityListStateInterface extends SynchronizationWrapper<ModuleIdentity[]>{}

export const moduleIndentityListStateInitial = Record<ModuleIndentityListStateInterface>({
    payload: [],
    synchronizationStatus: SynchronizationStatus.initialized
});

export type ModuleIndentityListStateType = IM<ModuleIndentityListStateInterface>;
