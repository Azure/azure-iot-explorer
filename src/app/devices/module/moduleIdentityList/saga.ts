/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import { call, put, takeLatest } from 'redux-saga/effects';
import { SagaIterator } from 'redux-saga';
import { Action } from 'typescript-fsa';
import { fetchModuleIdentities } from '../../../api/services/moduleService';
import { raiseNotificationToast } from '../../../notifications/components/notificationToast';
import { NotificationType } from '../../../api/models/notification';
import { ResourceKeys } from '../../../../localization/resourceKeys';
import { getModuleIdentitiesAction } from './actions';

export function* getModuleIdentitiesSagaWorker(action: Action<string>): SagaIterator {
    try {
        const parameters = {
            deviceId: action.payload
        };

        const moduleIdentities = yield call(fetchModuleIdentities, parameters);

        yield put(getModuleIdentitiesAction.done({params: action.payload, result: moduleIdentities}));
    } catch (error) {
        yield call(raiseNotificationToast, {
            text: {
                translationKey: ResourceKeys.notifications.getModuleIdentitiesOnError,
                translationOptions: {
                    deviceId: action.payload,
                    error,
                },
            },
            type: NotificationType.error
          });

        yield put(getModuleIdentitiesAction.failed({params: action.payload, error}));
    }
}

export function* getModuleIdentitiesSaga() {
    yield takeLatest(getModuleIdentitiesAction.started.type, getModuleIdentitiesSagaWorker);
}
