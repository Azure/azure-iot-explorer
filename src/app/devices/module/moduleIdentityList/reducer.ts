/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import { reducerWithInitialState } from 'typescript-fsa-reducers';
import { moduleIndentityListStateInitial, ModuleIndentityListStateType } from './state';
import { getModuleIdentitiesAction } from './actions';
import { SynchronizationStatus } from '../../../api/models/synchronizationStatus';
import { ModuleIdentity } from '../../../api/models/moduleIdentity';

export const moduleIdentityListReducer = reducerWithInitialState<ModuleIndentityListStateType>(moduleIndentityListStateInitial())
    .case(getModuleIdentitiesAction.started, (state: ModuleIndentityListStateType) => {
        return state.merge({
            synchronizationStatus: SynchronizationStatus.working
        });
    })
    .case(getModuleIdentitiesAction.done, (state: ModuleIndentityListStateType, payload: {params: string} & {result: ModuleIdentity[]}) => {
        return state.merge({
            payload: payload.result,
            synchronizationStatus: SynchronizationStatus.fetched
        });
    })
    .case(getModuleIdentitiesAction.failed, (state: ModuleIndentityListStateType) => {
        return state.merge({
            synchronizationStatus: SynchronizationStatus.failed
        });
    });
