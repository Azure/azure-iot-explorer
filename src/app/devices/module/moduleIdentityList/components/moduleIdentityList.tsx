/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import { useTranslation } from 'react-i18next';
import { NavLink, useLocation, useHistory, useRouteMatch } from 'react-router-dom';
import { IColumn, SelectionMode, CommandBar, Label } from '@fluentui/react';
import { ResizableDetailsList } from '../../../../shared/resizeDetailsList/resizableDetailsList';
import { ResourceKeys } from '../../../../../localization/resourceKeys';
import { getDeviceIdFromQueryString } from '../../../../shared/utils/queryStringHelper';
import { REFRESH, ArrayOperation } from '../../../../constants/iconNames';
import { SynchronizationStatus } from '../../../../api/models/synchronizationStatus';
import { parseDateTimeString } from '../../../../api/dataTransforms/transformHelper';
import { ModuleIdentity } from '../../../../api/models/moduleIdentity';
import { MultiLineShimmer } from '../../../../shared/components/multiLineShimmer';
import { ROUTE_PARTS, ROUTE_PARAMS } from '../../../../constants/routes';
import { HeaderView } from '../../../../shared/components/headerView';
import { useAsyncSagaReducer } from '../../../../shared/hooks/useAsyncSagaReducer';
import { moduleIndentityListStateInitial } from '../state';
import { getModuleIdentitiesSaga } from '../saga';
import { moduleIdentityListReducer } from '../reducer';
import { getModuleIdentitiesAction } from '../actions';
import { AppInsightsClient } from '../../../../shared/appTelemetry/appInsightsClient';
import { TELEMETRY_PAGE_NAMES } from '../../../../../app/constants/telemetry';
import '../../../../css/_deviceDetail.scss';

export const ModuleIdentityList: React.FC = () => {
    const { t } = useTranslation();
    const { search, pathname } = useLocation();
    const { url } = useRouteMatch();
    const history = useHistory();
    const deviceId = getDeviceIdFromQueryString(search);

    const [ localState, dispatch ] = useAsyncSagaReducer(moduleIdentityListReducer, getModuleIdentitiesSaga, moduleIndentityListStateInitial(), 'moduleIdentityListState');
    const moduleIdentityList = localState.payload;
    const synchronizationStatus = localState.synchronizationStatus;

    React.useEffect(() => {
        dispatch(getModuleIdentitiesAction.started(deviceId));
    },              [deviceId]);

    React.useEffect(() => {
        AppInsightsClient.getInstance()?.trackPageView({name: TELEMETRY_PAGE_NAMES.MODULE_LIST});
    }, []); // tslint:disable-line: align

    const showCommandBar = () => {
        return (
            <CommandBar
                className="command"
                items={[
                    {
                        ariaLabel: t(ResourceKeys.moduleIdentity.command.add),
                        iconProps: {iconName: ArrayOperation.ADD},
                        key: ArrayOperation.ADD,
                        name: t(ResourceKeys.moduleIdentity.command.add),
                        onClick: handleAdd
                    },
                    {
                        ariaLabel: t(ResourceKeys.moduleIdentity.command.refresh),
                        disabled: synchronizationStatus === SynchronizationStatus.working,
                        iconProps: {iconName: REFRESH},
                        key: REFRESH,
                        name: t(ResourceKeys.moduleIdentity.command.refresh),
                        onClick: handleRefresh
                    }
                ]}
            />
        );
    };

    const handleAdd = () => {
        const path = pathname.concat(ROUTE_PARTS.ADD);
        history.push(`${path}/?${ROUTE_PARAMS.DEVICE_ID}=${encodeURIComponent(deviceId)}`);
    };

    const handleRefresh = () => {
        dispatch(getModuleIdentitiesAction.started(deviceId));
    };

    const renderGrid = () => {
        return (
            <div className="device-detail">
                 <div className="list-detail">
                    <ResizableDetailsList
                        columns={getColumns()}
                        items={moduleIdentityList}
                        selectionMode={SelectionMode.none}
                        onRenderItemColumn={renderItemColumn}
                    />
                </div>

                {synchronizationStatus === SynchronizationStatus.working && <MultiLineShimmer/>}
                {synchronizationStatus === SynchronizationStatus.fetched && moduleIdentityList.length === 0 && t(ResourceKeys.moduleIdentity.noModules)}
                {synchronizationStatus === SynchronizationStatus.failed && t(ResourceKeys.moduleIdentity.errorFetching)}
            </div>
        );
    };

    const getModuleDetailPageUrl = (item: ModuleIdentity) => {
        return `${url.endsWith('/') ? url : url + '/'}${ROUTE_PARTS.MODULE_DETAIL}/?${ROUTE_PARAMS.DEVICE_ID}=${encodeURIComponent(deviceId)}&${ROUTE_PARAMS.MODULE_ID}=${item.moduleId}`;
    };

    // tslint:disable-next-line: cyclomatic-complexity
    const renderItemColumn = (item: ModuleIdentity, index: number, column: IColumn) => {
        switch (column.key) {
            case 'moduleId':
                return (
                <NavLink
                    key={column.key}
                    to={getModuleDetailPageUrl(item)}
                >
                    {item.moduleId}
                </NavLink>
                );
            case 'connectionState':
                return (
                    <Label
                        key={column.key}
                    >
                        {item.connectionState}
                    </Label>
                );
            case 'connectionStateLastUpdated':
                return (
                    <Label
                        key={column.key}
                    >
                        {parseDateTimeString(item.connectionStateUpdatedTime) || '--'}
                    </Label>
                );
            case 'lastActivityTime':
                return (
                    <Label
                        key={column.key}
                    >
                        {parseDateTimeString(item.lastActivityTime) || '--'}
                    </Label>
                );
            case 'modelId':
                return (
                    <Label
                        key={column.key}
                    >
                        {item.modelId || '--'}
                    </Label>
                );
            default:
                return <></>;
        }
    };

    const getColumns = (): IColumn[] => {
        const columns: IColumn[] = [
            {
                ariaLabel: t(ResourceKeys.moduleIdentity.columns.moduleId),
                fieldName: 'moduleId',
                key: 'moduleId',
                maxWidth: 200,
                minWidth: 50,
                name: t(ResourceKeys.moduleIdentity.columns.moduleId),
            },
            {
                ariaLabel: t(ResourceKeys.moduleIdentity.columns.connectionState),
                fieldName: 'connectionState',
                key: 'connectionState',
                maxWidth: 200,
                minWidth: 50,
                name: t(ResourceKeys.moduleIdentity.columns.connectionState)
            },
            {
                ariaLabel: t(ResourceKeys.moduleIdentity.columns.connectionStateLastUpdated),
                fieldName: 'connectionStateLastUpdated',
                key: 'connectionStateLastUpdated',
                maxWidth: 250,
                minWidth: 150,
                name: t(ResourceKeys.moduleIdentity.columns.connectionStateLastUpdated)
            },
            {
                ariaLabel: t(ResourceKeys.moduleIdentity.columns.lastActivityTime),
                fieldName: 'lastActivityTime',
                key: 'lastActivityTime',
                maxWidth: 250,
                minWidth: 150,
                name: t(ResourceKeys.moduleIdentity.columns.lastActivityTime)
            },
            {
                ariaLabel: t(ResourceKeys.moduleIdentity.columns.modelId),
                fieldName: 'modelId',
                key: 'modelId',
                maxWidth: 250,
                minWidth: 150,
                name: t(ResourceKeys.moduleIdentity.columns.modelId)
            }];

        return columns;
    };

    return (
        <>
            {showCommandBar()}
            <HeaderView
                headerText={ResourceKeys.moduleIdentity.headerText}
                link={ResourceKeys.moduleIdentity.link}
                tooltip={ResourceKeys.moduleIdentity.tooltip}
            />
            {renderGrid()}
        </>
    );
};
