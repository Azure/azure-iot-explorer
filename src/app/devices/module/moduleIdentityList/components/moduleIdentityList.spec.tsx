/***********************************************************
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License
 **********************************************************/
import * as React from 'react';
import 'jest';
import { shallow, mount } from 'enzyme';
import { CommandBar } from '@fluentui/react';
import { ModuleIdentityList } from './moduleIdentityList';
import { SynchronizationStatus } from '../../../../api/models/synchronizationStatus';
import * as AsyncSagaReducer from '../../../../shared/hooks/useAsyncSagaReducer';
import { getModuleIdentitiesAction } from '../actions';

const pathname = `/`;

jest.mock('react-router-dom', () => ({
    useHistory: () => ({ push: jest.fn() }),
    useLocation: () => ({ search: '', pathname }),
    useRouteMatch: () => ({ url: pathname }),
}));

describe('ModuleIdentityList', () => {
    context('snapshot', () => {
        it('matches snapshot while loading', () => {
            const initialState = {
                payload: [],
                synchronizationStatus: SynchronizationStatus.working
            };
            jest.spyOn(AsyncSagaReducer, 'useAsyncSagaReducer').mockReturnValueOnce([initialState, jest.fn()]);
            expect(shallow(<ModuleIdentityList/>)).toMatchSnapshot();
        });

        it('matches snapshot when fetch failed', () => {
            const initialState = {
                payload: [],
                synchronizationStatus: SynchronizationStatus.failed
            };
            jest.spyOn(AsyncSagaReducer, 'useAsyncSagaReducer').mockReturnValueOnce([initialState, jest.fn()]);
            expect(shallow(<ModuleIdentityList/>)).toMatchSnapshot();
        });

        it('matches snapshot with moduleIdentityList', () => {
            const initialState = {
                payload: [{
                    authentication: null,
                    deviceId: 'testDevice',
                    moduleId: 'testModule'
                }],
                synchronizationStatus: SynchronizationStatus.working
            };
            jest.spyOn(AsyncSagaReducer, 'useAsyncSagaReducer').mockReturnValueOnce([initialState, jest.fn()]);
            expect(shallow(<ModuleIdentityList/>)).toMatchSnapshot();
        });

        it('calls refresh', () => {
            const getModuleIdentitiesActionSpy = jest.spyOn(getModuleIdentitiesAction, 'started');
            const initialState = {
                payload: [],
                synchronizationStatus: SynchronizationStatus.fetched
            };
            jest.spyOn(AsyncSagaReducer, 'useAsyncSagaReducer').mockReturnValueOnce([initialState, jest.fn()]);
            const wrapper = mount(<ModuleIdentityList/>);
            const commandBar = wrapper.find(CommandBar).first();
            commandBar.props().items[1].onClick(null);
            expect(getModuleIdentitiesActionSpy).toBeCalled();
        });
    });
});
