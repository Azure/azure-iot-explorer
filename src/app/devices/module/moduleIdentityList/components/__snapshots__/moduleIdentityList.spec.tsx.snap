// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModuleIdentityList snapshot matches snapshot when fetch failed 1`] = `
<Fragment>
  <StyledCommandBarBase
    className="command"
    items={
      Array [
        Object {
          "ariaLabel": "moduleIdentity.command.add",
          "iconProps": Object {
            "iconName": "BoxAdditionSolid",
          },
          "key": "BoxAdditionSolid",
          "name": "moduleIdentity.command.add",
          "onClick": [Function],
        },
        Object {
          "ariaLabel": "moduleIdentity.command.refresh",
          "disabled": false,
          "iconProps": Object {
            "iconName": "Refresh",
          },
          "key": "Refresh",
          "name": "moduleIdentity.command.refresh",
          "onClick": [Function],
        },
      ]
    }
  />
  <HeaderView
    headerText="moduleIdentity.headerText"
    link="moduleIdentity.link"
    tooltip="moduleIdentity.tooltip"
  />
  <div
    className="device-detail"
  >
    <div
      className="list-detail"
    >
      <ResizableDetailsList
        columns={
          Array [
            Object {
              "ariaLabel": "moduleIdentity.columns.moduleId",
              "fieldName": "moduleId",
              "key": "moduleId",
              "maxWidth": 200,
              "minWidth": 50,
              "name": "moduleIdentity.columns.moduleId",
            },
            Object {
              "ariaLabel": "moduleIdentity.columns.connectionState",
              "fieldName": "connectionState",
              "key": "connectionState",
              "maxWidth": 200,
              "minWidth": 50,
              "name": "moduleIdentity.columns.connectionState",
            },
            Object {
              "ariaLabel": "moduleIdentity.columns.connectionStateLastUpdated",
              "fieldName": "connectionStateLastUpdated",
              "key": "connectionStateLastUpdated",
              "maxWidth": 250,
              "minWidth": 150,
              "name": "moduleIdentity.columns.connectionStateLastUpdated",
            },
            Object {
              "ariaLabel": "moduleIdentity.columns.lastActivityTime",
              "fieldName": "lastActivityTime",
              "key": "lastActivityTime",
              "maxWidth": 250,
              "minWidth": 150,
              "name": "moduleIdentity.columns.lastActivityTime",
            },
            Object {
              "ariaLabel": "moduleIdentity.columns.modelId",
              "fieldName": "modelId",
              "key": "modelId",
              "maxWidth": 250,
              "minWidth": 150,
              "name": "moduleIdentity.columns.modelId",
            },
          ]
        }
        items={Array []}
        onRenderItemColumn={[Function]}
        selectionMode={0}
      />
    </div>
    moduleIdentity.errorFetching
  </div>
</Fragment>
`;

exports[`ModuleIdentityList snapshot matches snapshot while loading 1`] = `
<Fragment>
  <StyledCommandBarBase
    className="command"
    items={
      Array [
        Object {
          "ariaLabel": "moduleIdentity.command.add",
          "iconProps": Object {
            "iconName": "BoxAdditionSolid",
          },
          "key": "BoxAdditionSolid",
          "name": "moduleIdentity.command.add",
          "onClick": [Function],
        },
        Object {
          "ariaLabel": "moduleIdentity.command.refresh",
          "disabled": true,
          "iconProps": Object {
            "iconName": "Refresh",
          },
          "key": "Refresh",
          "name": "moduleIdentity.command.refresh",
          "onClick": [Function],
        },
      ]
    }
  />
  <HeaderView
    headerText="moduleIdentity.headerText"
    link="moduleIdentity.link"
    tooltip="moduleIdentity.tooltip"
  />
  <div
    className="device-detail"
  >
    <div
      className="list-detail"
    >
      <ResizableDetailsList
        columns={
          Array [
            Object {
              "ariaLabel": "moduleIdentity.columns.moduleId",
              "fieldName": "moduleId",
              "key": "moduleId",
              "maxWidth": 200,
              "minWidth": 50,
              "name": "moduleIdentity.columns.moduleId",
            },
            Object {
              "ariaLabel": "moduleIdentity.columns.connectionState",
              "fieldName": "connectionState",
              "key": "connectionState",
              "maxWidth": 200,
              "minWidth": 50,
              "name": "moduleIdentity.columns.connectionState",
            },
            Object {
              "ariaLabel": "moduleIdentity.columns.connectionStateLastUpdated",
              "fieldName": "connectionStateLastUpdated",
              "key": "connectionStateLastUpdated",
              "maxWidth": 250,
              "minWidth": 150,
              "name": "moduleIdentity.columns.connectionStateLastUpdated",
            },
            Object {
              "ariaLabel": "moduleIdentity.columns.lastActivityTime",
              "fieldName": "lastActivityTime",
              "key": "lastActivityTime",
              "maxWidth": 250,
              "minWidth": 150,
              "name": "moduleIdentity.columns.lastActivityTime",
            },
            Object {
              "ariaLabel": "moduleIdentity.columns.modelId",
              "fieldName": "modelId",
              "key": "modelId",
              "maxWidth": 250,
              "minWidth": 150,
              "name": "moduleIdentity.columns.modelId",
            },
          ]
        }
        items={Array []}
        onRenderItemColumn={[Function]}
        selectionMode={0}
      />
    </div>
    <MultiLineShimmer />
  </div>
</Fragment>
`;

exports[`ModuleIdentityList snapshot matches snapshot with moduleIdentityList 1`] = `
<Fragment>
  <StyledCommandBarBase
    className="command"
    items={
      Array [
        Object {
          "ariaLabel": "moduleIdentity.command.add",
          "iconProps": Object {
            "iconName": "BoxAdditionSolid",
          },
          "key": "BoxAdditionSolid",
          "name": "moduleIdentity.command.add",
          "onClick": [Function],
        },
        Object {
          "ariaLabel": "moduleIdentity.command.refresh",
          "disabled": true,
          "iconProps": Object {
            "iconName": "Refresh",
          },
          "key": "Refresh",
          "name": "moduleIdentity.command.refresh",
          "onClick": [Function],
        },
      ]
    }
  />
  <HeaderView
    headerText="moduleIdentity.headerText"
    link="moduleIdentity.link"
    tooltip="moduleIdentity.tooltip"
  />
  <div
    className="device-detail"
  >
    <div
      className="list-detail"
    >
      <ResizableDetailsList
        columns={
          Array [
            Object {
              "ariaLabel": "moduleIdentity.columns.moduleId",
              "fieldName": "moduleId",
              "key": "moduleId",
              "maxWidth": 200,
              "minWidth": 50,
              "name": "moduleIdentity.columns.moduleId",
            },
            Object {
              "ariaLabel": "moduleIdentity.columns.connectionState",
              "fieldName": "connectionState",
              "key": "connectionState",
              "maxWidth": 200,
              "minWidth": 50,
              "name": "moduleIdentity.columns.connectionState",
            },
            Object {
              "ariaLabel": "moduleIdentity.columns.connectionStateLastUpdated",
              "fieldName": "connectionStateLastUpdated",
              "key": "connectionStateLastUpdated",
              "maxWidth": 250,
              "minWidth": 150,
              "name": "moduleIdentity.columns.connectionStateLastUpdated",
            },
            Object {
              "ariaLabel": "moduleIdentity.columns.lastActivityTime",
              "fieldName": "lastActivityTime",
              "key": "lastActivityTime",
              "maxWidth": 250,
              "minWidth": 150,
              "name": "moduleIdentity.columns.lastActivityTime",
            },
            Object {
              "ariaLabel": "moduleIdentity.columns.modelId",
              "fieldName": "modelId",
              "key": "modelId",
              "maxWidth": 250,
              "minWidth": 150,
              "name": "moduleIdentity.columns.modelId",
            },
          ]
        }
        items={
          Array [
            Object {
              "authentication": null,
              "deviceId": "testDevice",
              "moduleId": "testModule",
            },
          ]
        }
        onRenderItemColumn={[Function]}
        selectionMode={0}
      />
    </div>
    <MultiLineShimmer />
  </div>
</Fragment>
`;
